### DISCUSSION TITLE
Database test example?

### DISCUSSION THREAD

**Discussion by Flynsarmy** [dingo-d/wp-pest#18] [General]
Can we get an example in the readme of a DB-based test in the readme?

I tried something along the lines of the following:
```php
test("Adding a record to the db", function () {
    global $wpdb;
    $row = $wpdb->get_row("
        SELECT *
        FROM {$wpdb->postmeta}
        WHERE meta_id=6969
    ", ARRAY_A);
    expect($row)->not()->toBeEmpty();
})->with([
    function () {
        global $wpdb;
        $wpdb->insert($wpdb->postmeta, [
            'meta_id' => 6969,
            'post_id' => 9696,
            'meta_key' => 'testing',
            'meta_value' => 'Just a test.'
        ]);
    }
]);
```

While this worked the first time around, the DB wasn't reset the tests finished, so the next time I ran `vendor/bin/pest --group=integration` I got an Integrity constraint violation error as my test tried to write a record that already exists.

The DB not resetting after tests conclude is a problem, but it really should reset between individual tests too. How do we go about doing that?

A couple of examples in the readme would be greatly appreciated!

**Comment by dingo-d** [comment]
Oh this is a great information!

I'm on a vacation, but when I get back I'll give it a go.

I think that the db is reset after a db run, I will need to see how to do that after each test and maybe add a helper that can be called in the `afterEach` method.

**Comment by Flynsarmy** [comment]
In my testing I did find out that some tables are cleared between tests. For example all posts are deleted before tests run.

On a related note, this can cause problems if you're using a delete_post hook in your plugin because the plugin is loaded before delete_all_posts is automatically called resulting in unwanted side effects.

**Comment by dingo-d** [comment]
Opened issue: #29
